bee-network/src/peers/list.rs

```diff
@@ -0,0 +1,217 @@
+// Copyright 2020 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{Multiaddr, ShortId, MAX_UNKNOWN_PEERS};
+
+use super::{errors::Error, DataSender, PeerRelation};
+
+use libp2p::PeerId;
+use log::trace;
+use tokio::sync::RwLock;
+
+use std::{
+    collections::HashMap,
+    sync::{atomic::Ordering, Arc},
+};
+
+const DEFAULT_PEERLIST_CAPACITY: usize = 8;
+
+#[derive(Clone, Default)]
+pub struct PeerList(Arc<RwLock<HashMap<PeerId, (PeerInfo, PeerState)>>>);
```

Comment:
`BannedList` uses `DashSet` but `PeerList` uses `RwLock<HashMap>`? Is this due to the deadlocking issue?

Reply:
We are using the new 4.0 of dashmap, so the deadlock behaviour should be fixed if I am not mistaken now

Reply:
I switched to `RwLock<HashMap<_, _>>`, whereby RwLock is the async version from tokio. Until DashMap exhibits deadlocks (at it still does so even in version 4.0) I'ld rather not use it. Any objections to that?